src/main/java/usecases/RunCourseUseCase/RunCourseInteractor.java

```diff
@@ -0,0 +1,50 @@
+package usecases.RunCourseUseCase;
+
+import entities.Course;
+import entities.CourseFactory;
+import entities.Question;
+import java.util.List;
+import java.util.ArrayList;
+
+public class RunCourseInteractor {
+    private CourseFactory courseFactory;
+    private RunCourseSessionDataAccess sessionDbGateway;
+    private RunCourseOutputBoundary presenter;
+
+    public RunCourseInteractor(CourseFactory courseFactory, RunCourseSessionDataAccess sessionDbGateway, RunCourseOutputBoundary presenter){
+        this.courseFactory = courseFactory;
+        this.sessionDbGateway = sessionDbGateway;
+        this.presenter = presenter;
+    }
+    public boolean answersValid(List<String> answers) {
+        for (int i = 0; i < answers.size(); i++) {
+            if (answers.get(i).equals("")) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public String runCourse(RunCourseRequestModel requestModel) {
+        if (!answersValid(requestModel.getAnswers())){
+            return presenter.failView("Answers can not be empty.");
+        }
+
+        Course course = CourseFactory.create(requestModel.getId(), requestModel.getQuestions(), requestModel.getAnswers(), requestModel.getPoints());
+
+        List<Question> questions = course.getQuestions();
+        List<String> answers = new ArrayList<String>(questions.size());
+        List<Integer> points = new ArrayList<Integer>(questions.size());
+        for (int i = 0; i < questions.size(); i++){
+            Question question = course.getQuestionAt(i);
+            answers.add(question.getAnswer());
+            points.add(question.getPoints());
+        }
+
+        RunCourseDbRequestModel dbRequestModel = new RunCourseDbRequestModel(course.getCourseId(), course.getQuestions(), answers, points);
```

Comment:
I was at first wondering why this would be here. But I figured out why, I'm going to add the explanation below, and I think we should add a comment explaining it because otherwise we'll forget.

Since our Question class has both the answers and the points number. We can't pass in just the questions because otherwise we'll pass an entity to our gateway, which we should never ever do, this would violate CA. Gateways should never have knowledge about our entities. Thus we have to create a list of questions, answers, and points so our gateway can handle it.

I actually think this is redundant. We shouldn't create a course here, and we shouldn't pass in everything to the session gateway. Because after the CourseEnrollment Feature, we would assume that the entire course (without the answers) has been successfully added the the session database. Thus we only really require this interactor to give the answers to the session database since the session database will already have all the questions and points. 

I won't ask for changes because this version works and we need to merge all of our features soon. But we should definitely refactor this because in this case you only ever need the answers.